src/models/db.js

```diff
@@ -178,6 +181,29 @@ module.exports = config => {
       //   }
       // }
       //
+
+      // Given a POST to modelA with an inline entity modelB, the implicit
+      // association between modelA and modelB needs to be reflected in the
+      // inline entity, in order to avoid mandatory association issues.
+      try {
+        const modelA = instance.$modelOptions.name.plural;
+        const associationsOfModelB = association.target.associations;
+        Object.keys(associationsOfModelB).forEach(associationName => {
+          if (getModelName(associationName) === modelA) {
+            if (associatedEntity[associationName]) {
+              throw Object.create({
+                name: BAD_REQUEST,
+                errno: ERRNO_BAD_REQUEST,
+                errors: 'Could not create entity with body ' + associatedEntity
+              });
+            }
+            associatedEntity[associationName] = { '@iot.id': instance.id }
+          }
+        });
+      } catch(error) {
+        return Promise.reject(error);
```

Comment:
As a rule of thumb, do not trust on my code :P 

Yes, try removing both try/catch blocks, please. Let's stick to what the sequelize docs say.